src/test/java/io/confluent/idesidecar/restapi/integration/AbstractIT.java

```diff
@@ -0,0 +1,358 @@
+import io.confluent.idesidecar.restapi.util.TestEnvironment;
+import io.confluent.kafka.schemaregistry.client.CachedSchemaRegistryClient;
+import io.quarkus.logging.Log;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.serialization.ByteArrayDeserializer;
+import org.junit.jupiter.api.AfterEach;
+
+/**
+ * Abstract base class for integration tests that require a {@link TestEnvironment} and use
+ * a sidecar client to interact with the Sidecar REST API.
+ *
+ * <h2>Sharing the {@link TestEnvironment}</h2>
+ * <p>All subclasses of this class will share the same {@link LocalTestEnvironment} instance,
+ * which is started before any tests are run and stopped after all tests have run.
+ * Starting the test environment's containers takes 5-10 seconds, so doing it once
+ * <i>for all integration tests</i> (that extend this class) helps the tests run faster.
+ *
+ * <h2>Sharing or not sharing connections</h2>
+ * <p>Subclasses are expected to set up the sidecar connection(s) that they want their test
+ * methods to use. Creating a connection may take several seconds, depending upon the effort
+ * require to validate the connection and determine the status. This class provides a flexible
+ * way for subclasses to control how connections are created and reused.
+ *
+ * <p>Most subclasses will reuse the same connection for all tests. Reusing the same connection in
+ * multiple tests reduces the setup time for each test, as only the first test will need to create
+ * the connection. (The sidecar will be terminated after all tests in the test class are run,
+ * so deleting the connection is not necessary.)
+ *
+ * <p>Most of the {@link AbstractIT} concrete subclasses are defined so that all the tests will
+ * run against the same connection. Each test class should call the
+ * {@link #setupConnection(Class, Optional)} in it's {@code @BeforeEach} method to set up
+ * connection, passing the test class and a {@link ConnectionSpec} instance from the
+ * {@link TestEnvironment}, such as {@link TestEnvironment#localConnectionSpec()} or
+ * {@link TestEnvironment#directConnectionSpec()}.
+ *
+ * <p>Other tests may want to use a different connection for each test. If the test class is
+ * testing the Connections API, then the test class may want each <i>test method</i> to create
+ * its own connection(s) via the {@link #setupConnection(String, ConnectionSpec)} methods,
+ * verify the result of the creation, and when neceesary delete the connection.
+ * The {@code @AfterEach} method should still call {@link #deleteAllConnections()} to clean up all
+ * connections that remain at the end of each test, giving a clean slate to the next test method.
+ */
+public abstract class AbstractIT extends SidecarClient implements ITSuite {
```

Comment:
This is mostly from the `AbstractSidecarIT` that was renamed and relocated. But the `setupLocalConnection(...)` and `setupDirectConnection(...)` methods in `AbstractSidecarID` were replaced by one set of generic `setupConnection(...)` methods in this class that take the `TestEnvironment` and `ConnectionSpec` objects, various forms of which can be obtained from the `TestEnvironment` object. This makes it a bit easier to write the many different subclasses that each setup one connection.